services:
  db:
    image: postgres:15
    environment:
      POSTGRES_DB: social_media_db
      POSTGRES_USER: django_user
      POSTGRES_PASSWORD: strongpassword123
    volumes:
      - postgres_data:/var/lib/postgresql/data
    ports:
      - "5434:5432"
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U django_user -d social_media_db"]
      interval: 5s
      timeout: 5s
      retries: 5

  web:
    build: .
    command: >
      sh -c "python manage.py migrate &&
             python manage.py runserver 0.0.0.0:8000"
    volumes:
      - .:/app
    ports:
      - "8000:8000"
    depends_on:
      db:
        condition: service_healthy
    environment:
      - SECRET_KEY=django-insecure-q8-^dl6ya+%mc#x-pqsofg5^4s-oslbp5uuyen)+)ri07#xo&=
      - DEBUG=True
      - ALLOWED_HOSTS=localhost,web,127.0.0.1
      - DATABASE_NAME=social_media_db
      - DATABASE_USER=django_user
      - DATABASE_PASSWORD=strongpassword123
      - DATABASE_HOST=db
      - DATABASE_PORT=5432

volumes:
  postgres_data: